src/frontend/src/icp-eth/components/core/CkEthereumPendingTransactionsListener.svelte

```diff
@@ -3,54 +3,65 @@
 	import type { TransactionResponse } from 'ethers/providers';
 	import { onDestroy } from 'svelte';
 	import { initPendingTransactionsListener as initEthPendingTransactionsListenerProvider } from '$eth/providers/alchemy.providers';
-	import { tokenAsIcToken } from '$icp/derived/ic-token.derived';
 	import { icPendingTransactionsStore } from '$icp/stores/ic-pending-transactions.store';
-	import {
-		ckEthereumNativeTokenId,
-		ckEthereumTwinToken,
-		ckEthereumTwinTokenStandard
-	} from '$icp-eth/derived/cketh.derived';
+	import type { IcCkToken, IcToken } from '$icp/types/ic-token';
+	import { isIcCkToken } from '$icp/validation/ic-token.validation';
 	import {
 		loadPendingCkEthereumTransaction,
 		loadCkEthereumPendingTransactions
 	} from '$icp-eth/services/eth.services';
-	import { ckEthMinterInfoStore } from '$icp-eth/stores/cketh.store';
+	import { type CkEthMinterInfoData, ckEthMinterInfoStore } from '$icp-eth/stores/cketh.store';
 	import {
 		toCkErc20HelperContractAddress,
 		toCkEthHelperContractAddress
 	} from '$icp-eth/utils/cketh.utils';
 	import { ethAddress } from '$lib/derived/address.derived';
 	import { authIdentity } from '$lib/derived/auth.derived';
 	import { balance } from '$lib/derived/balances.derived';
-	import { tokenId } from '$lib/derived/token.derived';
-	import { token } from '$lib/stores/token.store';
 	import type { OptionEthAddress } from '$lib/types/address';
 	import type { OptionBalance } from '$lib/types/balance';
 	import type { WebSocketListener } from '$lib/types/listener';
-	import type { NetworkId } from '$lib/types/network';
+	import type { OptionToken, Token } from '$lib/types/token';
+	import type { CertifiedStoreData } from '$lib/stores/certified.store.js';
+	import type { BRAND } from 'zod';
+
+	export let token: OptionToken;
+	export let ckEthereumNativeToken: Token;
 
 	let listener: WebSocketListener | undefined = undefined;
 
 	let loadBalance: OptionBalance = undefined;
 
+	const twinToken = nonNullish(token) && isIcCkToken(token) ? token.twinToken : undefined;
```

Comment:
this should be reactive, i think

```suggestion
let twinToken: Token;
$: twinToken = nonNullish(token) && isIcCkToken(token) ? token.twinToken : undefined;
```